src/main/java/dev/warrant/WarrantClient.java

```diff
@@ -198,15 +199,12 @@ public WarrantCheckResult isAuthorized(WarrantCheck toCheck) throws WarrantExcep
         }
     }
 
-    public boolean hasPermission(String permissionId, String userId) throws WarrantException {
-        Subject userSubject = new Subject("user", userId);
-        Warrant permissionWarrant = new Warrant("permission", permissionId, "member", userSubject);
-        WarrantCheck permissionCheck = new WarrantCheck(Arrays.asList(permissionWarrant));
-        WarrantCheckResult result = isAuthorized(permissionCheck);
-        if (result.getCode() == 200) {
-            return true;
-        }
-        return false;
+    public boolean hasPermission(PermissionCheck permissionCheck) throws WarrantException {
+        Subject userSubject = new Subject("user", permissionCheck.getUserId());
+        Warrant permissionWarrant = new Warrant("permission", permissionCheck.getPermissionId(), "member", userSubject);
+        WarrantCheck warrantCheck = new WarrantCheck(Arrays.asList(permissionWarrant));
+        WarrantCheckResult result = isAuthorized(warrantCheck);
+        return result.getCode() == 200;
```

Comment:
Should this check the result of `WarrantCheckResult` instead?

Reply:
Checking either the `code` or the `result` string would work.

Reply:
I'll update it to check the `result` string